Source/WebCore/PAL/Configurations/PAL.xcconfig

```diff
@@ -67,5 +67,55 @@ DAV1D_LIBTOOLFLAGS_YES = $(BUILT_PRODUCTS_DIR)/libdav1d.a;
 
 
 SWIFT_OBJC_INTERFACE_HEADER_NAME = PALSwift.h;
-SWIFT_INSTALL_OBJC_HEADER = YES;
 SWIFT_VERSION = 5.0;
+
+SWIFT_OBJC_INTEROP_MODE = $(SWIFT_OBJC_INTEROP_MODE_$(WK_PLATFORM_NAME));
+SWIFT_OBJC_INTEROP_MODE_macosx = $(SWIFT_OBJC_INTEROP_MODE$(WK_MACOS_1400));
+SWIFT_OBJC_INTEROP_MODE_maccatalyst = $(SWIFT_OBJC_INTEROP_MODE$(WK_MACCATALYST_14));
+SWIFT_OBJC_INTEROP_MODE_iphoneos = $(SWIFT_OBJC_INTEROP_MODE$(WK_IOS_17));
+SWIFT_OBJC_INTEROP_MODE_iphonesimulator = $(SWIFT_OBJC_INTEROP_MODE$(WK_IOS_17));
+SWIFT_OBJC_INTEROP_MODE_appletvos = $(SWIFT_OBJC_INTEROP_MODE$(WK_TVOS_17));
+SWIFT_OBJC_INTEROP_MODE_appletvsimulator = $(SWIFT_OBJC_INTEROP_MODE$(WK_TVOS_17));
+SWIFT_OBJC_INTEROP_MODE_watchos = $(SWIFT_OBJC_INTEROP_MODE$(WK_WATCHOS_10));
+SWIFT_OBJC_INTEROP_MODE_watchsimulator = $(SWIFT_OBJC_INTEROP_MODE$(WK_WATCHOS_10));
+SWIFT_OBJC_INTEROP_MODE_visionos = objcxx;
+SWIFT_OBJC_INTEROP_MODE_visionsimulator = objcxx;
+SWIFT_OBJC_INTEROP_MODE_MACOS_SINCE_1400 = objcxx;
+SWIFT_OBJC_INTEROP_MODE_MACCATALYST_SINCE_14 = objcxx;
+SWIFT_OBJC_INTEROP_MODE_IOS_SINCE_17 = objcxx;
+SWIFT_OBJC_INTEROP_MODE_WATCHOS_SINCE_10 = objcxx;
+SWIFT_OBJC_INTEROP_MODE_TVOS_SINCE_17 = objcxx;
```

Comment:
Do we actually need to _avoid_ setting this on older platforms? If we aren't compiling swift sources on pre-2023 OS versions, then I would expect that this setting to have no effect. So we could just set it unconditionally.

Reply:
The generator code does trigger on every platform so I am just locking this down as much as possible.

Reply:
done.